"Add extensions" and "Debug" commands prompt for standard mode

datho7561 opened the pull request:
Add extensions and Debug are always visible in the command palette. If either Add extensions or Debug is selected while the Java server is in LightWeight mode, the user is prompted with a dialog that allows them to switch to Standard mode. The actions are performed after the switch is made.

This solution was adopted from [a vscode-java-debug commit that solves a similar issue](https://github.com/microsoft/vscode-java-debug/commit/20cd68cbb112f170b4b0aaf849604cf37853091a)

Fixes #299

Signed-off-by: David Thompson <davthomp@redhat.com>

angelozerr commented on package.json:
```diff
@@ -186,18 +186,6 @@
-      "commandPalette": [
-        {
-          "command": "quarkusTools.addExtension",
-          "when": "quarkusProjectExists"
```
If you remove this declaration where `quarkusProjectExists` condition is used?

angelozerr commented on src/extension.ts:
```diff
@@ -131,14 +132,22 @@ function registerVSCodeCommands(context: ExtensionContext) {
-    addExtensionsWizard();
+    requestStandardMode("Adding extensions").then((isStandardMode) => {
+      if (isStandardMode) {
+        addExtensionsWizard();
```
I think it misses a test with `quarkusProjectExists` context

angelozerr commented on src/extension.ts:
```diff
@@ -131,14 +132,22 @@ function registerVSCodeCommands(context: ExtensionContext) {
   context.subscriptions.push(commands.registerCommand(VSCodeCommands.DEBUG_QUARKUS_PROJECT, () => {
-    tryStartDebugging();
+    requestStandardMode("Debugging the project").then((isStandardMode) => {
+      if (isStandardMode) {
```
I think it misses a test with `quarkusProjectExists` context

datho7561 committed:
Debug and Add extension appear after switch

Debug and add extension are always visible in lightweight mode
if the extension is active. Using them prompts to enable standard mode.
After switching to standard mode:
 * If the project is not a Quarkus project, then an error appears
 * If the project is a Quarkus project, the command is run normally.

If vscode-java switches to standard mode through some other means,
or it is started in standard mode, then the debug and add extension
commands are shown/hidden depending on if the project is a Quarkus
project.

Fixes #299

Signed-off-by: David Thompson <davthomp@redhat.com>
